src/routes/safe/components/Apps/components/SignMessageModal/ReviewMessage.tsx

```diff
@@ -0,0 +1,262 @@
+import Divider from 'src/components/Divider'
+
+import { SignMessageModalProps } from '.'
+import Hairline from 'src/components/layout/Hairline'
+import { ButtonStatus, Modal } from 'src/components/Modal'
+import { grantedSelector } from 'src/routes/safe/container/selector'
+import Paragraph from 'src/components/layout/Paragraph'
+
+const { nativeCoin } = getNetworkInfo()
+
+const Container = styled.div`
+  max-width: 480px;
+  padding: ${md} ${lg} 0;
+`
+const TransactionFeesWrapper = styled.div`
+  background-color: ${({ theme }) => theme.colors.background};
+  padding: ${sm} ${lg};
+`
+
+const StyledBlock = styled(Block)`
+  background-color: ${({ theme }) => theme.colors.separator};
+  width: fit-content;
+  padding: 5px 10px;
+  border-radius: 3px;
+  margin: 4px 0 0 40px;
+
+  display: flex;
+
+  > :nth-child(1) {
+    margin-right: 5px;
+  }
+`
+
+const MessageTextArea = styled(MuiTextField)`
+  &.MuiFormControl-root {
+    padding-bottom: ${({ theme }) => theme.margin.md};
+  }
+`
+
+const InfoMessage = styled(Paragraph)`
+  display: flex;
+  align-items: center;
+
+  > span:first-child {
+    margin-right: ${({ theme }) => theme.margin.xs};
+  }
+`
+
+type Props = Omit<SignMessageModalProps, 'message' | 'isOpen'> & {
+  txData: string
+  txRecipient: string
+  utf8Message: string
+}
+
+export const ReviewMessage = ({
+  app,
+  safeAddress,
+  ethBalance,
+  safeName,
+  onUserConfirm,
+  onClose,
+  onTxReject,
+  requestId,
+  utf8Message,
+  txData,
+  txRecipient,
+}: Props): ReactElement => {
+  const dispatch = useDispatch()
+  const explorerUrl = getExplorerInfo(safeAddress)
+  const isOwner = useSelector(grantedSelector)
+
+  const [manualSafeTxGas, setManualSafeTxGas] = useState(0)
+  const [manualGasPrice, setManualGasPrice] = useState<string | undefined>()
+  const [manualGasLimit, setManualGasLimit] = useState<string | undefined>()
+
+  const {
+    gasLimit,
+    gasPriceFormatted,
+    gasEstimation,
+    isOffChainSignature,
+    isCreation,
+    isExecution,
+    gasCostFormatted,
+    txEstimationExecutionStatus,
+  } = useEstimateTransactionGas({
+    txData: txData || '',
+    txRecipient,
+    operation: Operation.DELEGATE,
+    txAmount: '0',
+    safeTxGas: manualSafeTxGas,
+    manualGasPrice,
+    manualGasLimit,
+  })
+
+  const [buttonStatus, setButtonStatus] = useEstimationStatus(txEstimationExecutionStatus)
+
+  const handleTxRejection = () => {
+    onTxReject(requestId)
+    onClose()
+  }
+
+  const handleUserConfirmation = (safeTxHash: string): void => {
+    onUserConfirm(safeTxHash, requestId)
+    onClose()
+  }
+
+  const confirmTransactions = (txParameters: TxParameters) => {
+    setButtonStatus(ButtonStatus.LOADING)
+
+    dispatch(
+      createTransaction(
+        {
+          safeAddress,
+          to: txRecipient,
+          valueInWei: '0',
+          txData,
+          operation: Operation.DELEGATE,
+          origin: app.id,
+          navigateToTransactionsTab: false,
```

Comment:
Is it intended that this is `false`?

Reply:
Confirmed that is always `false` for now.

Reply:
Some context for other reviewers: this was copy-pasted from there https://github.com/gnosis/safe-react/blob/dev/src/routes/safe/components/Apps/components/ConfirmTxModal/ReviewConfirm.tsx#L163